### PR TITLE
frontend/feat/#1063: added push notification for clevertap for xiaomi…

### PR DESCRIPTION
**Pull request by ashkriti**
… and huawei

## Type of Change

- [ ] Bugfix
- [x] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Integrated CleverTap push notification for Xiaomi and Huawei

## Motivation and Context
https://github.com/nammayatri/nammayatri/issues/1063

## How did you test it?
Tested Manually

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/android-native/app/libs/MiPush_SDK_Client_5_1_5-G_3rd.aar
Comment (kavya-shree-s) [inline Frontend/android-native/app/libs/MiPush_SDK_Client_5_1_5-G_3rd.aar:1]: @ashkriti Why aar file is required, Looking like all the sdks are from remote

**Review by kavya-shree-s — CHANGES_REQUESTED** [review]

File: Frontend/android-native/app/libs/MiPush_SDK_Client_5_1_5-G_3rd.aar
Comment (ashkriti) [reply Frontend/android-native/app/libs/MiPush_SDK_Client_5_1_5-G_3rd.aar:1]: It is required to configure our app with CleverTap Xiaomi plugin according to their document

File: Frontend/android-native/mobility-app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -61,7 +61,27 @@
...
+            </intent-filter>
+            <intent-filter>
+                <action android:name="com.xiaomi.mipush.ERROR" />
+            </intent-filter>
+        </receiver>
+        <service
+            android:name=".MyHmsMessageService"
+            android:exported="false">
+            <intent-filter>
+                <action android:name="com.huawei.push.action.MESSAGING_EVENT" />
```
Comment (Vignesh-772) [inline Frontend/android-native/mobility-app/src/main/AndroidManifest.xml:82]: these two permissions are not added 
```
    <permission
        android:name="yourpackage.permission.MIPUSH_RECEIVE"
        android:protectionLevel="signature" />
    <uses-permission android:name="yourpackage.permission.MIPUSH_RECEIVE" />
    <uses-permission android:name="android.permission.VIBRATE" />
```

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]

File: Frontend/android-native/mobility-app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (ashkriti) [reply Frontend/android-native/mobility-app/src/main/AndroidManifest.xml:82]: It is already added in the .aar file

**Review by Vignesh-772 — DISMISSED** [review]

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/MyHmsMessageService.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+import androidx.annotation.Nullable;
+import com.clevertap.android.sdk.CleverTapAPI;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Iterator;
+
+public class MyHmsMessageService extends HmsMessageService {
```
Comment (Rohit4997) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/MyHmsMessageService.java:16]: Please Rename `MyHmsMessageService`

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

**Review by Rohit4997 — APPROVED** [review]

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/HmsMessageService.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
+package in.juspay.mobility.app;
```
Comment (Rohit4997) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/HmsMessageService.java:1]: Add Copyright

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/XiaomiMessageReceiver.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
+package in.juspay.mobility.app;
```
Comment (Rohit4997) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/XiaomiMessageReceiver.java:1]: Add Copyright

Comment (ashkriti) [reply Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/XiaomiMessageReceiver.java:10]: added

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/HmsMessageService.java (hunk 3 above)
Comment (ashkriti) [reply Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/HmsMessageService.java:10]: added

**Review by Rohit4997 — APPROVED** [review]

**Review by kavya-shree-s — APPROVED** [review]

**Pull request merged**